scripts/dev/dump_diagnostic.py

```diff
@@ -0,0 +1,120 @@
+import os
+import json
+import yaml
+import sys
+import typing
+
+from kubernetes.client.rest import ApiException
+from kubernetes import client
+
+
+def clean_nones(value):
+    """
+    Recursively remove all None values from dictionaries and lists, and returns
+    the result as a new dictionary or list.
+    """
+    if isinstance(value, list):
+        return [clean_nones(x) for x in value if x is not None]
+    if isinstance(value, dict):
+        return {key: clean_nones(val) for key, val in value.items() if val is not None}
+    return value
+
+
+def header(msg: str) -> str:
+    dashes = (
+        "----------------------------------------------------------------------------"
+    )
+    return "\n{0}\n{1}\n{0}\n".format(dashes, msg)
+
+
+def dump_crd(crd_log: typing.TextIO):
+    crdv1 = client.ApiextensionsV1beta1Api()
+    try:
+        crd_log.write(header("CRD"))
+        mdb = crdv1.list_custom_resource_definition(pretty="true")
+        crd_log.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_custom_resource_definition: %s\n" % e)
+
+
+def dump_persistent_volume(diagnosticFile: typing.TextIO):
+    corev1 = client.CoreV1Api()
+    try:
+        diagnosticFile.write(header("Persistent Volume Claims"))
+        mdb = corev1.list_persistent_volume(pretty="true")
+        diagnosticFile.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_persistent_volume %s\n" % e)
+
+
+def dump_stateful_sets_namespaced(diagnosticFile: typing.TextIO, namespace: str):
+    av1beta1 = client.AppsV1Api()
+    try:
+        diagnosticFile.write(header("Stateful Sets"))
+        mdb = av1beta1.list_namespaced_stateful_set(namespace, pretty="true")
+        diagnosticFile.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_namespaced_stateful_set: %s\n" % e)
+
+
+def dump_pod_log_namespaced(namespace: str, name: str):
```

Comment:
I think this function is a bit specific to our use case, I think it will always be safe to just grab all the pods in a given namespace, and if any pod has multiple containers, we can just grab the logs of those too. I don't think we need to explicitly look for agent/mongod/operator names.